dev_tools/scripts/create_new_algorithm.sh

```diff
@@ -231,11 +231,11 @@ then
         for ((i=1; i<=${#CMAKE_FILES[@]}; i++))
         do
             sourcepath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${TEMPLATE_FILES[$i-1]}
-            for j in $(ls ${GINKGO_ROOT_DIR}/${sourcepath})
+        for j in "${GINKGO_ROOT_DIR}/${sourcepath}/"*;
```

Comment:
Same here, the `ls` is really needed.

Reply:
Does it work here without the quoted `${sourcepath}`

Reply:
Weird. at least `for j in "${GINKGO_ROOT_DIR}/"${sourcepath}/*;` worked for me and tobias.
There should be an alternative, because ls has some cases in which it is unsafe.

Reply:
We may keep the ls for now. But should keep the problem with ls in consideration. This could break when some things like spaces occur in files.

I'll leave the ls in there then

Reply:
And the indentation was wrong. yes.
Luckily that doesn't matter in bash. Still needs to be changed though